lib/fluent/plugin/out_tdlog.rb

```diff
@@ -216,6 +216,8 @@ def format_stream(tag, es)
         begin
           record.to_msgpack(out)
         rescue RangeError
+          # msgpack v0.4 appends broken record to out so it should be removed
+          out = out[0, off] unless $use_msgpack_5
```

Comment:
this needs to be `out = out.to_s[0, off]` (remove `unless $use_msgpack_5`) because v5 also causes the same problem depending on the data size:

```
irb(main):006:0> out = ''
irb(main):007:0> ["large star"*10000, 100000000000000000000000000000].to_msgpack(out)
RangeError: bignum too big to convert into `unsigned long long'
	from (irb):7:in `to_msgpack'
	from (irb):7
	from /Users/frsyuki/.rbenv/versions/1.9.3-p194/bin/irb:12:in `<main>'
irb(main):008:0> out
=> "\x92\xDB\x00\x01\x86\xA0"
```

```
irb(main):009:0> out = ''
irb(main):010:0> ["short str", 100000000000000000000000000000].to_msgpack(out)
RangeError: bignum too big to convert into `unsigned long long'
	from (irb):10:in `to_msgpack'
	from (irb):10
	from /Users/frsyuki/.rbenv/versions/1.9.3-p194/bin/irb:12:in `<main>'
irb(main):011:0> out
=> ""
```